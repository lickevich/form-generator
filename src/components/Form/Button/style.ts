import styled from "styled-components";

export const StyledInput = styled.input`
  margin-left: 10px;
  margin-bottom: 10px;
  padding: 6px 12px;
  cursor: pointer;
`;
