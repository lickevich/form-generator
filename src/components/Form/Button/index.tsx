import React from "react";

import { StyledInput } from "./style";
import { IButton } from "../../../type";

const Button: React.FC<IButton> = ({ type, value, ...rest }) => (
  <StyledInput type={type} value={value} {...rest} />
);

export default Button;
