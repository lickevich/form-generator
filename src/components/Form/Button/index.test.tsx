import React from "react";
import { render } from "@testing-library/react";

import Button from "./";

describe("Button", () => {
  it("render the component", () => {
    const component = <Button type="button" value="OK" />;
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
