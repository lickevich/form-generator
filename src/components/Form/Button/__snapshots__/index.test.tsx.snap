// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render the component 1`] = `
<DocumentFragment>
  <input
    class="sc-bdvvaa hkAxcz"
    type="button"
    value="OK"
  />
</DocumentFragment>
`;
